### PR TITLE
ci: add lighthouse CI

### PR DESCRIPTION
**Pull request by krshdev**
This is not the final copy of the action probably, this is just to get started, we can improve it as we use :D

@eddiejaoude you need to do this before this PR can be merged: https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/getting-started.md#github-app-method-recommended

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/927"></a>

### CODE REVIEW

**Comment by emmalearnscode** [comment]
This repo has the action I mentioned on the livestream. You also need to add a config file though and allow lighthouse ci access to the repo. I’m not sure how the two compare, but mine stops a deploy if scores fall below a certain level. We maybe need to decide if that is something we want or if lighthouse scores are just a nice to have that we can keep an eye on and raise issues ourselves if values fall below a certain level. https://github.com/emmalearnscode/learn-cool-problems/blob/main/lighthouserc.json

**Review by emmalearnscode — CHANGES_REQUESTED** [review]
As Cahl has a PR open to update node version to 16, it’s probably a good idea that this one matches too.

File: .github/workflows/lighthouse.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Lighthouse CI
+on: [push]
+jobs:
+  lhci:
+    name: Lighthouse
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 12.x
```
Comment (krishguptadev) [inline .github/workflows/lighthouse.yml:9]:
```suggestion
      - name: Use Node.js 16.x
```

File: .github/workflows/lighthouse.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
```
Comment (krishguptadev) [inline .github/workflows/lighthouse.yml:12]:
```suggestion
          node-version: 16.x
```

**Comment by krshdev** [comment]
Hi @emmalearnscode, could you review this once? 🙂

File: .github/workflows/lighthouse.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: Lighthouse CI
+on: [push]
+jobs:
+  lhci:
+    name: Lighthouse
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v1
```
Comment (nhcarrigan) [inline .github/workflows/lighthouse.yml:10]:
```suggestion
        uses: actions/setup-node@v3
```
v1 is very outdated.

**Review by nhcarrigan — CHANGES_REQUESTED** [review]

**Comment by krshdev** [comment]
#1152 (Sorry, I think I deleted the branch)

**Pull request closed without merging**